build_tools/github_actions/test_executable_scripts/test_hipsparse.py

```diff
@@ -9,13 +9,25 @@
 SCRIPT_DIR = Path(__file__).resolve().parent
 THEROCK_DIR = SCRIPT_DIR.parent.parent.parent
 
+# GTest sharding
+SHARD_INDEX = os.getenv("SHARD_INDEX")
+TOTAL_SHARDS = os.getenv("TOTAL_SHARDS")
+envion_vars = os.environ.copy()
+# For display purposes in the GitHub Action UI, the shard array is 1th indexed. However for shard indexes, we convert it to 0th index.
+envion_vars["GTEST_SHARD_INDEX"] = str(int(SHARD_INDEX) - 1)
+envion_vars["GTEST_TOTAL_SHARDS"] = str(TOTAL_SHARDS)
```

Comment:
Can you add defaults for these `getenv` calls, so the scripts can be run locally with the variables not set? I'm a bit worried about the increasing complexity in these test scripts, since it should be easy enough to read the script and see how to run the tests directly on a development machine not through the scripts.